runtime/development/src/xcm.rs

```diff
@@ -319,9 +320,36 @@ pub type LocationToAccountId = (
 	// Sibling parachain origins convert to AccountId via the `ParaId::into`.
 	SiblingParachainConvertsVia<Sibling, AccountId>,
 	// Straight up local `AccountId32` origins just alias directly to `AccountId`.
+	// TODO(nuno): We probably should drop this because this allows for the relay-chain
+	// to impersonate a local, native user.
 	AccountId32Aliases<RelayNetwork, AccountId>,
+	AccountIdHash<AccountId>,
 );
 
+use codec::Encode;
+use sp_io::hashing::blake2_256;
+use sp_std::borrow::Borrow;
+
+// A wildcard to convert a MultiLocation to a Centrifuge AccountId
+pub struct AccountIdHash<AccountId>(PhantomData<AccountId>);
+impl<AccountId: From<[u8; 32]> + Into<[u8; 32]> + Clone>
+	xcm_executor::traits::Convert<MultiLocation, AccountId> for AccountIdHash<AccountId>
+{
+	fn convert_ref(location: impl Borrow<MultiLocation>) -> Result<AccountId, ()> {
+		let hash: [u8; 32] = ("multiloc", location.borrow())
+			.borrow()
+			.using_encoded(blake2_256);
+		let account_id: AccountId = hash.into();
+		log::trace!(target: "xcm::convert_origin", "AccountIdHash with location {:?} converting to AccountId {}", location.borrow(), hex::encode(hash));
+
+		Ok(account_id)
+	}
+
+	fn reverse_ref(_: impl Borrow<AccountId>) -> Result<MultiLocation, ()> {
+		Err(())
+	}
+}
+
```

Comment:
I will move this to a new `xcm-primitives` crate with a bunch of other pieces that can be shared across all the different runtime's xcm configurations

Reply:
Seems that there is an implementation already that hashes a location into an account, `location_conversion::Account32Hash`. Is there any reason why we do not use that one?

Reply:
Thanks for the catch @mikiquantum, I didn't realised there was an implementation already 👍  The official one expects a `Network: Get<NetworkId>` type param that is not used in the end, which is off, but it's fine. Done ✅